lib/core/rlocks.nim

```diff
@@ -48,7 +48,7 @@ proc release*(lock: var RLock) {.inline.} =
   ## Releases the given lock.
   releaseSys(lock)
 
-template withRLock*(lock: var RLock, code: untyped): untyped =
+template withRLock*(lock: RLock, code: untyped) =
```

Comment:
Huh? But the other template was changed to have the `untyped` return type!

Reply:
sorry, I closed my PR. I found an example that's not threadsafe. https://github.com/nim-lang/Nim/pull/18413#issuecomment-872127845